application_test.go

```diff
@@ -427,3 +428,21 @@ func TestApplicationConfiguration(t *testing.T) {
 	assert.True(t, ok)
 	assert.Equal(t, ErrCodeNotFound, apiErr.ErrCode)
 }
+
+func TestWaitOnApplication(t *testing.T) {
+	endpoint := newFakeMarathonEndpoint(t, nil)
+	defer endpoint.Close()
+
+	err := endpoint.Client.WaitOnApplication(fakeAppName, 0*time.Millisecond)
```

Comment:
I'm wondering if this approach carries a risk for flakiness: It basically relies on the timeout error returning before the result of `TestWaitOnApplication`'s does. On my local machine, the test would always succeed; I'm not sure though if a different machine or system behavior could make a change.

Any thoughts on this one?